tests/test.cpp

```diff
@@ -29,6 +29,23 @@
 #endif
 // clang-format on
 
+  // Ensure we are robust against adversarial namespaces:
+  namespace MyGame {
```

Comment:
Doesn't this test things already sufficiently? What is covered in the adversarial tests dir that this doesn't do?

Reply:
I put this in later. The adversarial-namespaces tests are likely to fail for one reason, but this is likely to fail for many reasons.

This file is probably better, because it tests more possible failures. I think that this case doesn't test custom string and pointer types, whereas the adversarial-namespaces tests do include those tests.

Reply:
The test here is very cheap, the one in the adversarial tests dir are more expensive. If the ones here cover the most likely cases (of something that is already rare), I'd suggest removing the test dir. I know its not fun to remove things you already made, but if it helps maintainability going forward, I think its better to keep things simple.